lib/launcher/src/Cardano/Launcher.hs

```diff
@@ -190,14 +197,39 @@ withBackendProcess tr cmd@(Command name args before output) action = do
         | name `isPrefixOf` show e = Just (ProcessDidNotStart name e)
         | otherwise = Nothing
 
+    -- Wraps 'waitForProcess' in another thread. This works around the unwanted
+    -- behaviour of the process library on Windows where 'waitForProcess' seems
+    -- to block all concurrent async actions in the thread.
+    interruptibleWaitForProcess
+        :: Trace IO LauncherLog
+        -> ProcessHandle
+        -> IO ExitCode
+    interruptibleWaitForProcess tr' ph = do
+        status <- newEmptyMVar
+        void $ forkIO $ waitThread status `onException` continue status
```

Comment:
I think Duncan's variant used `finally` rather than `onException` here which has a slightly different meaning (`finally` will also execute the action in case of normal exits).

Reply:
This will likely be used only for starting the backend / server which should never actually exit normally, but we are creating a potential deadlock by doing this. Is it intentional @rvl ?

Reply:
I needed to change it to use onException because we are capturing exit status. Otherwise it would try to putMVar twice

Reply:
Hmmmm. I see okay, there's actually a `putMVar` in `waitThread` itself which I guess _is safe_ (provided the `launcherLog` doesn't throw. But I am ready to accept anything at this stage).

Reply:
I think that because the putMVar is the last action in the block, either way there will be something put.